src/sympc/protocol/fss/fss.py

```diff
@@ -108,7 +108,12 @@ def evaluate(session: Session, b, x_masked, op, dtype="long") -> ShareTensor:
     dtype_options = {None: th.long, "int": th.int32, "long": th.long}
     result = th.tensor(result_share, dtype=dtype_options[dtype])
 
-    share_result = ShareTensor(data=result, session=session)
+    # stdlib
+    import dataclasses
+
+    share_result = ShareTensor(
+        data=result, session_uuid=session.uuid, **dataclasses.asdict(session.config)
```

Comment:
I think you do not have to make it a dictionary and unpack the config here, you can pass it to the `config` arg, because the items inside (`encoder_base` and `encoder_precision`) are no longer part of the init.